module/PurchaseArray.java

```diff
@@ -0,0 +1,46 @@
+package module;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+public class PurchaseArray {
+    private final AbstractPurchase[] purchases;
+
+    public PurchaseArray(AbstractPurchase[] purchases) {
+        this.purchases = purchases;
+    }
+
+    public void sort() {
+        Arrays.sort(purchases);
+    }
+
+    public int search( Euro searchValue) {
+        Comparator<AbstractPurchase> comparator = AbstractPurchase::compareTo;
+
+        return Arrays.binarySearch(
+                purchases,
+                new GreaterDiscount(new Product(null, new Euro(searchValue)), 1, 0.10),
+                comparator);

```

Comment:
```suggestion
        return Arrays.binarySearch(
                purchases,
                new GreaterDiscount(new Product(null, new Euro(searchValue)), 1, 0.10),
                );
```
`AbstractPurchase` has the internal comparator.